src/util/irep2_type.h

```diff
@@ -565,6 +593,47 @@ class array_type2t : public array_type_methods
   static std::string field_names[esbmct::num_type_fields];
 };
 
+/** Array type.
+ *  Comes with a subtype of the array and a size that might be constant, might
+ *  be nondeterministic, might be infinite. These facts are recorded in the
+ *  array_size and size_is_infinite fields.
+ *
+ *  If size_is_infinite is true, array_size will be null. If array_size is
+ *  not a constant number, then it's a dynamically sized array.
+ *  @extends array_data
+ */
+class vector_type2t : public vector_type_methods
+{
+public:
+  /** Primary constructor.
+   *  @param subtype Type of elements in this array.
+   *  @param size Size of this array.
+   *  @param inf Whether or not this array is infinitely sized
+   */
+  vector_type2t(const type2tc &_subtype, const expr2tc &size, bool inf)
```

Comment:
can vector be infinite sized? If not, remove this parameter and pass `false` to base class constructor.